pandas/tests/io/json/test_pandas.py

```diff
@@ -82,489 +80,379 @@ def setup(self, datapath):
         del self.tsframe
         del self.mixed_frame
 
-    def test_frame_double_encoded_labels(self):
+    def test_frame_double_encoded_labels(self, orient):
         df = DataFrame(
             [["a", "b"], ["c", "d"]],
             index=['index " 1', "index / 2"],
             columns=["a \\ b", "y / z"],
         )
 
-        assert_frame_equal(df, read_json(df.to_json(orient="split"), orient="split"))
-        assert_frame_equal(
-            df, read_json(df.to_json(orient="columns"), orient="columns")
-        )
-        assert_frame_equal(df, read_json(df.to_json(orient="index"), orient="index"))
-        df_unser = read_json(df.to_json(orient="records"), orient="records")
-        assert_index_equal(df.columns, df_unser.columns)
-        tm.assert_numpy_array_equal(df.values, df_unser.values)
+        result = read_json(df.to_json(orient=orient), orient=orient)
+        expected = df.copy()
+
+        if orient == "records" or orient == "values":
+            expected = expected.reset_index(drop=True)
+        if orient == "values":
+            expected.columns = range(len(expected.columns))
 
-    def test_frame_non_unique_index(self):
+        assert_frame_equal(result, expected)
+
+    @pytest.mark.parametrize(
+        "orient,expected",
+        [
+            (
+                "split",
+                DataFrame([["a", "b"], ["c", "d"]], index=[1, 1], columns=["x", "y"]),
+            ),
+            ("records", DataFrame([["a", "b"], ["c", "d"]], columns=["x", "y"])),
+            ("values", DataFrame([["a", "b"], ["c", "d"]])),
+        ],
+    )
+    def test_frame_non_unique_index(self, orient, expected):
         df = DataFrame([["a", "b"], ["c", "d"]], index=[1, 1], columns=["x", "y"])
 
-        msg = "DataFrame index must be unique for orient='index'"
-        with pytest.raises(ValueError, match=msg):
-            df.to_json(orient="index")
-        msg = "DataFrame index must be unique for orient='columns'"
+        result = read_json(df.to_json(orient=orient), orient=orient)
+        assert_frame_equal(result, expected)
+
+    @pytest.mark.parametrize("orient", ["index", "columns"])
+    def test_frame_non_unique_index_raises(self, orient):
+        df = DataFrame([["a", "b"], ["c", "d"]], index=[1, 1], columns=["x", "y"])
+        msg = "DataFrame index must be unique for orient='{}'".format(orient)
         with pytest.raises(ValueError, match=msg):
-            df.to_json(orient="columns")
+            df.to_json(orient=orient)
 
-        assert_frame_equal(df, read_json(df.to_json(orient="split"), orient="split"))
-        unser = read_json(df.to_json(orient="records"), orient="records")
-        tm.assert_index_equal(df.columns, unser.columns)
-        tm.assert_almost_equal(df.values, unser.values)
-        unser = read_json(df.to_json(orient="values"), orient="values")
-        tm.assert_numpy_array_equal(df.values, unser.values)
```

Comment:
111-116 looks similar to 92-99, which became part of 90-98.  Is this block subsumed into that, or did it go somewhere else?

Reply:
92-99 is part of test `test_frame_double_encoded_labels` whereas this is part of `test_frame_non_unique_index`. Each should be parametrized in its own test in the new module - does that help clarify where this was moved?

Reply:
Yes, thanks.  I think I mixed up test_frame_non_unique_index with test_frame_non_unique_index_raises.